### PR TITLE
Add endogenous export

### PR DESCRIPTION
**Pull request by energyLS**
# Closes # (if applicable).

## Changes proposed in this Pull Request
This PR adds the possibility of endogenous exports to the model. Previously, it was only possible to set a fixed volume of hydrogen exports (e.g. 10 TWh). This feature enables the possibility, to export hydrogen which is covered by a certain market price for hydrogen set in the config.

In detail, it adds these options:

- endogenous: A boolean flag indicating whether the export demand is endogenously determined by the model. If true, the export demand is determined by the model.
- endogenous_price: The market price (in EUR/MWh) at which hydrogen for endogenous exports is sold. This parameter is only considered if endogenous is set to true.

## Checklist

- [x] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `doc/requirements.txt`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by energyLS** [comment]
@ekatef I have added this PR and the config options are added in `config.default.yaml`. Do I also need to add the options in the `config.tutorial.yaml` or  `test/` for specific tests, or is that not required anymore? What do we test and what not?

@hazemakhalek I think you have previously come up with a similar feature in https://github.com/pypsa-meets-earth/pypsa-earth-sec/compare/main...bright. Can you point me to it, so I can outline the differences? Also, what do you think of this proposal?

File: config.default.yaml
Comment (ekatef) [inline config.default.yaml]: Could it probably make sense to group `export_profile` parameter together with other parameters which are being considered with `endogenous: false`, with `h2export`, in particular? My feling is that it would make the fragment a bit easier to comprehend

File: scripts/add_export.py
Comment (ekatef) [inline scripts/add_export.py]: Since `snakemake.params.export_endogenous` we should be able to use it directly as a condition, without `== False`

File: scripts/add_export.py
Comment (ekatef) [inline scripts/add_export.py]: A comparison with `True` seemse to be redundand

**Review by ekatef — COMMENTED** [review]
Great @energyLS! Sounds like a very nice enhancement 😄 
Have added a couple of comments to the code, which are quite minor though.

As for the configs, `config.tutorial` currently contains a power only model, and no updates are needed here. Regarding test configurations, it would be great to add a new option to `test/config.test_myopic.yaml`. The reason is that testing configs are build upon the tutorial config, so all the cross-sectoral parameters must be specified in the test configuration directly.

Could you also please update a version of the config in the [tutorial](https://github.com/pypsa-meets-earth/pypsa-earth/blob/aea9d27b0c1dde8bfb5b9a83c15c5b225f3940b0/config.tutorial.yaml#L5) and the [default](https://github.com/pypsa-meets-earth/pypsa-earth/blob/aea9d27b0c1dde8bfb5b9a83c15c5b225f3940b0/config.default.yaml#L5) configs?

**Comment by energyLS** [comment]
Dear @ekatef, thanks for reviewing! I have:

- Revised your suggestions to streamline the if-else condition: 0d67965a628edeaf0c2b97cc7ed4193385b09656
- Grouped endogenous and exogenous parameters together: cd0fd4d3e10d656ff3eeda355d88ee50966f2184. Or do you mean it could be grouped in a sense, that all exogenous/endogenous params are a subparameter? E.g. ["endogenous"]["endogenous_price"] and ["exogenous"]["h2export"] and so on?
- Updated the versions of `config.default.yaml` and `test/config.test_myopic.yaml`: c6c7ebe40eedf68041987a711d913fe9cfb8d06d
- Added the options to the test/config.test_myopic.yaml: f76b16fffabe76d8364522e73928ad3bf7329508. Should the new option be set to true, to actually test it? Or do we keep it as it is (endogenous=False) and "only" test on our standard configuration?

And since we haven't documented the config options of the sector-coupled version in https://pypsa-earth.readthedocs.io/en/latest/configuration.html yet, I guess there is nothing to add here at this point? Of course it should be done in general at one point :)

**Comment by ekatef** [comment]
> https://pypsa-earth.readthedocs.io/en/latest/configuration.html



> Dear @ekatef, thanks for reviewing! I have:
> 
> * Revised your suggestions to streamline the if-else condition: [0d67965](https://github.com/pypsa-meets-earth/pypsa-earth/commit/0d67965a628edeaf0c2b97cc7ed4193385b09656)
> * Grouped endogenous and exogenous parameters together: [cd0fd4d](https://github.com/pypsa-meets-earth/pypsa-earth/commit/cd0fd4d3e10d656ff3eeda355d88ee50966f2184). Or do you mean it could be grouped in a sense, that all exogenous/endogenous params are a subparameter? E.g. ["endogenous"]["endogenous_price"] and ["exogenous"]["h2export"] and so on?
> * Updated the versions of `config.default.yaml` and `test/config.test_myopic.yaml`: [c6c7ebe](https://github.com/pypsa-meets-earth/pypsa-earth/commit/c6c7ebe40eedf68041987a711d913fe9cfb8d06d)
> * Added the options to the test/config.test_myopic.yaml: [f76b16f](https://github.com/pypsa-meets-earth/pypsa-earth/commit/f76b16fffabe76d8364522e73928ad3bf7329508). Should the new option be set to true, to actually test it? Or do we keep it as it is (endogenous=False) and "only" test on our standard configuration?
> 
> And since we haven't documented the config options of the sector-coupled version in https://pypsa-earth.readthedocs.io/en/latest/configuration.html yet, I guess there is nothing to add here at this point? Of course it should be done in general at one point :)

Hey @energyLS, thanks a lot for implementing the changes! 

As a technical comment, not need to specify the particular commits. We trust you! 😉 

Agree that it feels worth to have an advanced version tested for each functionality, while I'm also confident in your expertise for sector-couple modeling. To me, the test configuration looks good as it is now.

Regarding documenting the config options for the sector-coupled part you are indeed absolutely right. That is clearly out of scope for this PR, but it would be definitely useful to open an issue on that. Would you mind doing that? 🙏🏽 

The only very minor point left is to update [the config version](https://github.com/pypsa-meets-earth/pypsa-earth/blob/e386102552dc4471115432b5980e08125238b5f7/config.tutorial.yaml#L5) also in the tutorial config. Also, please feel free to add a release note! That is quite a significant update, and it would be great to have it mentioned in the release.

After that I think the PR is ready to merge, unless @davide-f or @hazemakhalek have any comments.

**Comment by hazemakhalek** [comment]
I trust you both. Go ahead if you feel it's ready

**Comment by energyLS** [comment]
> > https://pypsa-earth.readthedocs.io/en/latest/configuration.html
> 
> > Dear @ekatef, thanks for reviewing! I have:
> > 
> > * Revised your suggestions to streamline the if-else condition: [0d67965](https://github.com/pypsa-meets-earth/pypsa-earth/commit/0d67965a628edeaf0c2b97cc7ed4193385b09656)
> > * Grouped endogenous and exogenous parameters together: [cd0fd4d](https://github.com/pypsa-meets-earth/pypsa-earth/commit/cd0fd4d3e10d656ff3eeda355d88ee50966f2184). Or do you mean it could be grouped in a sense, that all exogenous/endogenous params are a subparameter? E.g. ["endogenous"]["endogenous_price"] and ["exogenous"]["h2export"] and so on?
> > * Updated the versions of `config.default.yaml` and `test/config.test_myopic.yaml`: [c6c7ebe](https://github.com/pypsa-meets-earth/pypsa-earth/commit/c6c7ebe40eedf68041987a711d913fe9cfb8d06d)
> > * Added the options to the test/config.test_myopic.yaml: [f76b16f](https://github.com/pypsa-meets-earth/pypsa-earth/commit/f76b16fffabe76d8364522e73928ad3bf7329508). Should the new option be set to true, to actually test it? Or do we keep it as it is (endogenous=False) and "only" test on our standard configuration?
> > 
> > And since we haven't documented the config options of the sector-coupled version in https://pypsa-earth.readthedocs.io/en/latest/configuration.html yet, I guess there is nothing to add here at this point? Of course it should be done in general at one point :)
> 
> Hey @energyLS, thanks a lot for implementing the changes!
> 
> As a technical comment, not need to specify the particular commits. We trust you! 😉
> 
> Agree that it feels worth to have an advanced version tested for each functionality, while I'm also confident in your expertise for sector-couple modeling. To me, the test configuration looks good as it is now.
> 
> Regarding documenting the config options for the sector-coupled part you are indeed absolutely right. That is clearly out of scope for this PR, but it would be definitely useful to open an issue on that. Would you mind doing that? 🙏🏽
> 
> The only very minor point left is to update [the config version](https://github.com/pypsa-meets-earth/pypsa-earth/blob/e386102552dc4471115432b5980e08125238b5f7/config.tutorial.yaml#L5) also in the tutorial config. Also, please feel free to add a release note! That is quite a significant update, and it would be great to have it mentioned in the release.
> 
> After that I think the PR is ready to merge, unless @davide-f or @hazemakhalek have any comments.

Dear @ekatef, 
thanks for the feedback! I have added the new feature to the release notes, and adjusted the config version of the tutorial config. Also, I have set up a new issue for the documentation: https://github.com/pypsa-meets-earth/pypsa-earth/issues/1208

I think everthing is done now and ready to merge, however the CI currently fails due to a problem not related with this PR:
https://github.com/pypsa-meets-earth/pypsa-earth/actions/runs/12046200910/job/33586189025?pr=1201#step:15:5.
Once that is fixed, we're good to go!

**Review by ekatef — APPROVED** [review]

**Comment by ekatef** [comment]
> Dear @ekatef, thanks for the feedback! I have added the new feature to the release notes, and adjusted the config version of the tutorial config. Also, I have set up a new issue for the documentation: #1208
> 
> I think everthing is done now and ready to merge, however the CI currently fails due to a problem not related with this PR: https://github.com/pypsa-meets-earth/pypsa-earth/actions/runs/12046200910/job/33586189025?pr=1201#step:15:5. Once that is fixed, we're good to go!

Hey @energyLS, perfect! Thanks a lot for the changes, and opening the issue on the documentation is highly appreciated 😄 

Agree on the plan: let's fix the CI and merge.

**Comment by ekatef** [comment]
Hey @energyLS CI has been fixed thanks to the recent work by @davide-f!! 🎉 
Feel free to update your branch and re-run the workflows 😄

**Pull request merged**
